apps/opencs/view/render/terraintexturemode.cpp

```diff
@@ -0,0 +1,328 @@
+#include "terraintexturemode.hpp"
+#include "editmode.hpp"
+
+#include <string>
+
+#include <QWidget>
+#include <QIcon>
+#include <QSpinBox>
+#include <QGroupBox>
+#include <QSlider>
+#include <QEvent>
+#include <QDropEvent>
+#include <QButtonGroup>
+#include <QHBoxLayout>
+#include <QVBoxLayout>
+#include <QDragEnterEvent>
+#include <QDrag>
+
+#include "../widget/modebutton.hpp"
+#include "../widget/scenetoolbar.hpp"
+
+#include "pagedworldspacewidget.hpp"
+#include "mask.hpp"
+#include "object.hpp" // Something small needed regarding pointers from here ()
+#include "worldspacewidget.hpp"
+
+#include "../../model/doc/document.hpp"
+#include "../../model/world/columnbase.hpp"
+#include "../../model/world/commandmacro.hpp"
+#include "../../model/world/commands.hpp"
+#include "../../model/world/data.hpp"
+#include "../../model/world/idtable.hpp"
+#include "../../model/world/land.hpp"
+#include "../../model/world/landtexture.hpp"
+#include "../../model/world/resourcetable.hpp"
+#include "../../model/world/tablemimedata.hpp"
+#include "../../model/world/universalid.hpp"
+
+
+#include <components/esm/loadland.hpp>
```

Comment:
We normally include components before anything from the apps directory. And stuff from the local directory (the one the current file is in) is typically included last (except for including the matching hpp file in a cpp file of course; that always comes first). Functionally it doesn't really matter, but it makes the code a bit easier to read.